sfe/sfe.go

```diff
@@ -0,0 +1,317 @@
+	if len(filename) == 0 {
+		http.Error(w, "Template page does not exist", http.StatusInternalServerError)
+		return
+	}
+
+	err := sfe.templatePages.ExecuteTemplate(w, filename, dynamicData)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+	}
+}
+
+// Index is the homepage of the SFE
+func (sfe *SelfServiceFrontEndImpl) Index(response http.ResponseWriter, request *http.Request) {
+	if request.Method != http.MethodGet && request.Method != http.MethodHead {
+		response.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+	sfe.renderTemplate(response, "index.html", nil)
+}
+
+// BuildID tells the requester what boulder build version is running.
+func (sfe *SelfServiceFrontEndImpl) BuildID(response http.ResponseWriter, request *http.Request) {
+	response.Header().Set("Content-Type", "text/plain")
+	response.WriteHeader(http.StatusOK)
+	detailsString := fmt.Sprintf("Boulder=(%s %s)", core.GetBuildID(), core.GetBuildTime())
+	if _, err := fmt.Fprintln(response, detailsString); err != nil {
+		sfe.log.Warningf("Could not write response: %s", err)
+	}
+}
+
+// unpauseJWT is generated by a WFE and is used to round-trip back through the
+// SFE to unpause the requester's account.
+type unpauseJWT string
+
+func (sfe *SelfServiceFrontEndImpl) getHelper(response http.ResponseWriter, incomingJWT unpauseJWT) {
+	if incomingJWT != "" {
+		// Serve the actual unpause page given to a Subscriber. Populates the
+		// unpause form with the JWT from the URL. That JWT itself may be
+		// invalid or expired, but that validation will be performed after
+		// submitting the form.
+		sfe.unpausePrepareToUnpause(response, unpausePostForm, incomingJWT)
+	} else {
+		// We only want to accept requests containing the JWT param.
+		sfe.unpauseInvalidRequest(response)
+	}
+}
+
+// posthelper serves a page indicating if the unpause succeeded or failed upon
+// clicking the unpause button. We are explicitly choosing to not address CSRF
+// at this time because we control creation and redemption of the JWT.
+func (sfe *SelfServiceFrontEndImpl) postHelper(response http.ResponseWriter, request *http.Request, incomingJWT unpauseJWT) {
+	if incomingJWT != "" {
+		regID, _, err := sfe.validateUnpauseJWTforAccount(incomingJWT)
+		if err != nil {
+			sfe.unpauseFailure(response)
+			return
+		}
+
+		// TODO(#7356) Declare a registration ID variable to populate an
+		// rapb unpause account request message.
+		_, innerErr := strconv.ParseInt(regID, 10, 64)
+		if innerErr != nil {
+			sfe.unpauseFailure(response)
+			return
+		}
+
+		// TODO(#7536) Send gRPC nrequest to the RA informing it to unpause
+		// the account specified in the claim. At this point we should wait
+		// for the RA to process the request before returning to the client,
+		// just in case the request fails.
+
+		// Success, the account has been unpaused.
+		http.Redirect(response, request, unpauseSuccessful, http.StatusFound)
+	} else {
+		sfe.unpauseInvalidRequest(response)
```

Comment:
Same here -- getting to the POST url (which they should only arrive at via our own button!) without a JWT feels like an error case. Let's leave this as-is for now, but file a bug+TODO to improve the messaging here.